lib/scope_file_access.rb

```diff
@@ -17,18 +17,18 @@ def file_path(system_file)
     File.join(scope_file_store.path, system_file.name)
   end
 
-  def file_content(filename)
+  def file_content(system_file)
     if !extracted
       raise Machinery::Errors::FileUtilsError, "The requested file '#{filename}' is not available" \
         " because files for scope '#{scope_name}' were not extracted."
     end
 
-    path = File.join(scope_file_store.path, filename)
-    if !File.exist?(path)
-      raise Machinery::Errors::FileUtilsError, "The requested file '#{filename}' was not found."
-    end
+    File.read(system_file.scope.file_path(system_file))
```

Comment:
Since the module is included directly in the `Scope` object this can also be `File.read(file_path(system_file))` instead.